### PR TITLE
Fix circleci repo paths, etc.

### PR DESCRIPTION
**Pull request by MocanaAtBroad**
Discovered that after a recent tweak paths were messed. Main symptom was that are build-utils scripts could not be found because PATH was not set properly.
Then found that we were getting errors when trying to rebuild the Angular node modules. Needed to include some additional programs installed in supporting Docker image.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,8 +1,12 @@
 version: 2.1
 
 references:
+  repo_path: &repo_path
+      /home/circleci/repo
+  ng_workspace_path: &ng_workspace_path
+      /home/circleci/repo/ddp-workspace
```
Comment (MocanaAtBroad) [inline .circleci/config.yml:7]: Setting these up as " constants" is what we are doing in the ddp-study-server. Can be referenced wherever needed.

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
   npm_cache_key: &npm_cache_key
-    v1-dependency-npm1-{{ checksum "package.json" }}-{{ checksum "package-lock.json" }}
+    v1-dependency-npm2-{{ checksum "package.json" }}-{{ checksum "package-lock.json" }}
   npm_cache_restore_key: &npm_cache_restore_key
```
Comment (MocanaAtBroad) [inline .circleci/config.yml:10]: node_modules will now live in different absolute path, so old caches will not help us. So tweaked base key so we can refer to caches stored in new location.

File: .circleci/config.yml [hunk 3]
```diff
@@ -16,10 +20,10 @@ references:
 
 # Container for all the builds
 executors:
-  commit-executor:
+  build-executor:
     docker:
-      - image: broadinstitute/study-website-build:04-09-2020A
-    working_directory: ~/repo/ddp-workspace
+      - image: broadinstitute/study-website-build:05-06-2020A
```
Comment (MocanaAtBroad) [inline .circleci/config.yml:25]: New image. Already pushed to Docker hub.

File: .circleci/config.yml [hunk 4]
```diff
@@ -16,10 +20,10 @@ references:
...
+    working_directory: *ng_workspace_path
     resource_class: medium+
```
Comment (MocanaAtBroad) [inline .circleci/config.yml:27]: Using the new "constant" workspace path location. ddp-workspace is where most of the action occurs, so might as well make it the default.

File: .circleci/config.yml [hunk 5]
```diff
@@ -30,6 +34,7 @@ commands:
         type: string
     steps:
       - setup-build-workspace
+      - set-deployment-environment
```
Comment (MocanaAtBroad) [inline .circleci/config.yml:37]: Turns out we need to set deployment environment as this is where we get the build params now that determine whether we include source code or not in builds.

File: .circleci/config.yml [hunk 6]
```diff
@@ -114,7 +119,6 @@ commands:
             fi
             TAR_FILE_PATH="/tmp/$(basename -- $TAR_FILE_URL)"
             gsutil cp  $TAR_FILE_URL $TAR_FILE_PATH
-            cd ~/repo/ddp-workspace
```
Comment (MocanaAtBroad) [inline .circleci/config.yml:117]: It's the default already.

File: build-utils/Dockerfile-BuildDeploy [hunk 7]
```diff
@@ -114,6 +114,8 @@ RUN apk  --no-cache add \
     ruby \
     ruby-bundler \
     ruby-json  \
+    make \
+    g++ \
     && rm -rf /var/cache/apk/*
```
Comment (MocanaAtBroad) [inline build-utils/Dockerfile-BuildDeploy:119]: npm install sometimes wants to build some binaries. These are the programs it wants to use to do so.

**Review by zyme — APPROVED** [review]

**Review by fat23cat — APPROVED** [review]

**Pull request merged**